xwiki-platform-core/xwiki-platform-url/xwiki-platform-url-default/src/main/java/org/xwiki/url/internal/DefaultURLSecurityManager.java

```diff
@@ -214,6 +220,11 @@ public boolean isURITrusted(URI uri)
     public URI parseToSafeURI(String serializedURI) throws URISyntaxException, SecurityException
     {
         URI uri;
+        if (serializedURI.contains(PERCENT_ESCAPE)) {
+            throw new IllegalArgumentException(
+                String.format("The given uri [%s] contains the string [%s] which is used internally "
+                + "for performing escaping operations. Please use another marker.", serializedURI, PERCENT_ESCAPE));
```

Comment:
Wouldn't it make more sense to only perform this check when we're actually using this pattern, i.e., in the catch clause below?

Reply:
Right, it avoid a limitation for URI that we can parse with URI single constructor...